iothub/service/src/Amqp/AmqpReceivingLinkHandler.cs

```diff
@@ -97,16 +97,30 @@ internal async Task CloseAsync(CancellationToken cancellationToken)
         internal async Task AcknowledgeMessageAsync(ArraySegment<byte> deliveryTag, Outcome outcome, CancellationToken cancellationToken)
         {
             if (Logging.IsEnabled)
-                Logging.Enter(this, $"Acknowledging message with delivery tag {deliveryTag} on receiving link with address {_linkAddress} and link name {_linkName}", nameof(AcknowledgeMessageAsync));
+                Logging.Enter(
+                    this,
+                    $"Acknowledging message with delivery tag {deliveryTag} on receiving link with address {_linkAddress} and link name {_linkName}",
+                    nameof(AcknowledgeMessageAsync));
 
             try
             {
                 await _receivingLink.DisposeMessageAsync(deliveryTag, outcome, false, cancellationToken).ConfigureAwait(false);
             }
+            catch (AmqpException ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(
+                        this,
+                        $"Failed to acknowledge message with delivery tag {deliveryTag} on receiving link with address {_linkAddress} and link name {_linkName} with error {ex.Error} due to {ex}",
```

Comment:
I'd consider pointing out in this log statement that this will lead to the message being delivered again by the service later.